skcosmo/feature_selection/voronoi_fps.py

```diff
@@ -0,0 +1,229 @@
+            progress_bar=progress_bar,
+            initialize=initialize,
+            tolerance=tolerance,
+        )
+
+    def _init_greedy_search(self, X, y, n_to_select):
+        """Initializes the search. Prepares an array to store the selected
+        features. This function also determines the switching point if it was not
+        given during initialization."""
+
+        n_features = X.shape[1]
+        """index of the voronoi cell associated with each of the columns of X"""
+        self.vlocation_of_idx = np.full(n_features, -1)
+
+        """quarter of the square distance between new selected point and previously
+        selected points"""
+        self.sel_d2q_ = np.zeros(self.n_features_to_select, float)
+        self.new_dist_ = np.zeros(n_features)
+
+        """ Determines the optimal switching point for full calculation.
+        The point is that when calculating distances in Voronoi_FPS it is necessary
+        to "jump" between indexes in the array. This results in a significant increase in time.
+        Therefore, if you need to recalculate a large number of distances, it is more advantageous
+        to run Simple_FPS"""
+        if self.full_fraction is None:
+            simple_fps_timing = -time()
+            if not isinstance(self.n_trial_calculation_, numbers.Integral):
+                raise TypeError("Number of trial calculation should be integer")
+            if self.n_trial_calculation_ == 0:
+                raise ValueError(
+                    "Number of trial calculation should be more or equal to 1"
+                )
+            for i in range(self.n_trial_calculation_):
+                _ = X.T @ X[:, 0]
+            simple_fps_timing += time()
+            simple_fps_timing /= self.n_trial_calculation_
+
+            lower_fraction = 0
+            top_fraction = 1
+            while top_fraction - lower_fraction > 0.01:
+                voronoi_fps_times = np.zeros(self.n_trial_calculation_)
+                self.full_fraction = (top_fraction + lower_fraction) / 2
+                for i in range(self.n_trial_calculation_):
+                    sel = np.random.randint(
+                        n_features, size=int(n_features * self.full_fraction)
+                    )
+                    voronoi_fps_times[i] = -time()
+                    _ = X[:, 0] @ X[:, sel]
+                    voronoi_fps_times[i] += time()
+                voronoi_fps_timing = np.sum(voronoi_fps_times)
+                voronoi_fps_timing /= self.n_trial_calculation_
+                if voronoi_fps_timing < simple_fps_timing:
+                    lower_fraction = self.full_fraction
+                else:
+                    top_fraction = self.full_fraction
+
+            self.full_fraction = lower_fraction
+        else:
+            if isinstance(self.full_fraction, numbers.Real):
+                if not 0 < self.full_fraction <= 1:
+                    raise ValueError(
```

Comment:
could you add a small test for this error condition?

Reply:
I agree

Reply:
Yep, I added it. Indirectly, bit it check this case